src/playback/internal/playbackcontroller.cpp

```diff
@@ -777,12 +777,18 @@ void PlaybackController::addTrack(const InstrumentTrackId& instrumentTrackId, co
     }
 
     const std::string primaryInstrId = part->instrument()->id().toStdString();
+    if (instrumentTrackId.instrumentId == primaryInstrId) {
+        const std::string trackName = part->partName().toStdString();
+        doAddTrack(instrumentTrackId, trackName, onFinished);
+        return;
+    }
 
-    const std::string trackName = (instrumentTrackId.instrumentId == primaryInstrId)
-                                  ? part->partName().toStdString()
-                                  : "(" + part->instrumentById(instrumentTrackId.instrumentId)->trackName().toStdString() + ")";
-
-    doAddTrack(instrumentTrackId, trackName, onFinished);
+    const InstrumentTrackIdSet instrumentTrackIdSet = part->instrumentTrackIdSet();
```

Comment:
Note that in the implementation of `instrumentTrackIdSet()`, the set is constructed from a `std::map` (namely an InstrumentList), so there is some complexity there ($O(n\log(n))$ if I'm not mistaken). 
And then there is some additional complexity to check if the instrument is in the set, which is another $O(\log(n))$.
And then you additionally call `instrumentById`, which again iterates over `_instruments` inside the part.

It looks like you could avoid the use of `instrumentTrackIdSet` and `find` by simply checking if `instrumentById` returns nullptr. Doesn't that work?